lib/libc/shell.c

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/sys/printk.h>
+#include <zephyr/shell/shell.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <kernel_internal.h>
+#if defined(CONFIG_LOG_RUNTIME_FILTERING)
+#include <zephyr/logging/log_ctrl.h>
+#endif
+
+#include <malloc.h>
+
+static int cmd_app_heap(const struct shell *sh, size_t argc, char **argv)
+{
+	struct mallinfo mi = mallinfo();
+
+	ARG_UNUSED(argc);
+	ARG_UNUSED(argv);
+
+	shell_fprintf(sh, SHELL_NORMAL, "Heap size: %d bytes\n", mi.arena);
```

Comment:
This is fine but `shell_print` macro looks more nifty ;). You will not need `SHELL_NORMAL` and `\n`.